modules/mono/glue/GodotSharp/GodotSharp/Core/Extensions/NodeExtensions.cs

```diff
@@ -113,15 +117,20 @@ public partial class Node
         /// Negative indices access the children from the last one.
         /// To access a child node via its name, use <see cref="GetNode"/>.
         /// </summary>
-        /// <seealso cref="GetChild{T}(int)"/>
+        /// <seealso cref="GetChild{T}(int, bool)"/>
         /// <param name="idx">Child index.</param>
+        /// <param name="includeInternal">
+        /// If <see langword="false"/>, internal children are skipped (see <c>internal</c>
+        /// parameter in <see cref="AddChild(Node, bool, InternalMode)"/>).
+        /// </param>
         /// <typeparam name="T">The type to cast to. Should be a descendant of <see cref="Node"/>.</typeparam>
         /// <returns>
         /// The child <see cref="Node"/> at the given index <paramref name="idx"/>, or <see langword="null"/> if not found.
         /// </returns>
-        public T GetChildOrNull<T>(int idx) where T : class
+        public T GetChildOrNull<T>(int idx, bool includeInternal = false) where T : class
         {
-            return GetChild(idx) as T;
+            int count = GetChildCount();
+            return idx > -count && idx < count ? GetChild(idx, includeInternal) as T : null;
```

Comment:
`-count` is the first element index, `-1` last.
```suggestion
            return idx >= -count && idx < count ? GetChild(idx, includeInternal) as T : null;
```